middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -1327,7 +1327,9 @@ and make_switch ~tag_discriminant env res e arms =
           i + 1, res)
         arms (0, res)
     in
-    wrap (C.transl_switch_clambda Debuginfo.none e index cases), res
+    (* CR-someday poechsel: Put a correct value kind here *)
+    ( wrap (C.transl_switch_clambda Debuginfo.none (Vval Pgenval) e index cases),
```

Comment:
This comment should probably say "put a more precise value kind here" since the choice of `Pgenval` is correct but not necessarily the most precise.

The reasoning for it being correct is also subtle and deserves a comment. Something like:
```ocaml
(* The cases of the switch must have a value kind (i.e. they cannot be unboxed). This is because each case will either end in a `Cexit` or end by "calling" the return continuation, in which case it will just return the argument to that continuation. Currently functions cannot return unboxed values, so that argument must have a value kind. *)
```